TaskModules/powershell/Azure/SPNCreation.ps1

```diff
@@ -10,9 +10,100 @@ param
     [string] $spnRole = "owner",
     
     [Parameter(Mandatory=$false, HelpMessage="Provide Azure environment name for your subscription")]
-    [string] $environmentName = "AzureCloud"
+    [string] $environmentName = "AzureCloud",
+
+    [Parameter(Mandatory=$false, HelpMessage="Provide AzureStackManagementURL to add AzureStack environment to AzureRmEnvironments.")]
+    [string] $azureStackManagementURL
 )
 
+$AZURESTACK_ENVIRONMENT = "AzureStack"
+
+function Add-AzureStackToAzureRmEnvironment {
+    param (
+        [Parameter(mandatory=$true, HelpMessage="The Admin ARM endpoint URI of the Azure Stack Environment")]
+        $EndpointURI,
+        [parameter(mandatory=$true, HelpMessage="Azure Stack environment name for use with AzureRM commandlets")]
+        [string] $Name
+    )
+
+    $EndpointURI = $EndpointURI.TrimEnd("/")
+
+    $Domain = ""
+    try {
+        $uriendpoint = [System.Uri] $EndpointURI
+        $i = $EndpointURI.IndexOf('.')
+        $Domain = ($EndpointURI.Remove(0,$i+1)).TrimEnd('/')
+    }
+    catch {
+        Write-Error (Get-VstsLocString -Key AZ_InvalidARMEndpoint)
+    }
+
+    $ResourceManagerEndpoint = $EndpointURI
+    $stackdomain = $Domain
+
+    $AzureKeyVaultDnsSuffix="vault.$($stackdomain)".ToLowerInvariant()
+    $AzureKeyVaultServiceEndpointResourceId= $("https://vault.$stackdomain".ToLowerInvariant())
+    $StorageEndpointSuffix = ($stackdomain).ToLowerInvariant()
+
+
+    $azureStackEndpointUri = $EndpointURI.ToString() + "/metadata/endpoints?api-version=2015-01-01"
+
+    Write-Verbose "Retrieving endpoints from the $ResourceManagerEndpoint"
+
+    $endpointData = Invoke-RestMethod -Uri $azureStackEndpointUri -Method Get -ErrorAction Stop

```

Comment:
Do we need to catch the exception here and provide user friendly message?

Reply:
Catching the exception in the caller of this method.